src/geodesy/ntv2_grid.rs

```diff
@@ -0,0 +1,150 @@
+use js_sys::DataView;
+
+const SEC_TO_DEG: f64 = 0.0002777778;
+const SEC_PER_MIN: f64 = 60.0;
+
+// Both overview and sub grid headers have 11 fields of 16 bytes each.
+const NTV2_HEADER_SIZE: usize = 11 * 16;
+const NTV2_NODE_SIZE: usize = 16;
+
+// Buffer offsets for the NTv2 subgrid header
+const NTV2_SUBGRID_NLAT: usize = 88; // (f64) upper latitude (in seconds)
+const NTV2_SUBGRID_SLAT: usize = 72; // (f64) lower latitude (in seconds)
+const NTV2_SUBGRID_ELON: usize = 104; // (f64) lower longitude (in seconds)
+const NTV2_SUBGRID_WLON: usize = 120; // (f64) upper longitude (in seconds)
+const NTV2_SUBGRID_DLAT: usize = 136; // (f64) Latitude interval (in seconds)
+const NTV2_SUBGRID_DLON: usize = 152; // (f64) Longitude interval (in seconds)
+const NTV2_SUBGRID_GSCOUNT: usize = 168; // (i32) grid node count
+
+const NTV2_NODE_LAT_CORRN: usize = 0; // (f32) correction to the latitude at this node point (secs)
+const NTV2_NODE_LON_CORRN: usize = 4; // (f32) correction to the longitude at this node point (secs)
+
+/// Read a NTv2 grid from a `js_sys::DataView` into Gravsoft binary native to Rust Geodesy.
+pub fn parse_ntv2_to_gravsoft_bin(view: &DataView) -> Result<Vec<u8>> {
+    let is_le = view.get_int32_endian(8, true) == 11;
+
+    // TODO: read the header magic (string) and compare it to NTV2_MAGIC
+    let num_of_fields = view.get_int32_endian(8, is_le);
+    if num_of_fields != 11 {
+        return Err(Error::InvalidNtv2GridFormat("Wrong header").into());
+    }
+
+    let num_sub_grids = view.get_int32_endian(40, is_le) as usize;
+
+    if num_sub_grids != 1 {
+        // Multi grid support is out of scope for geodesy-wasm
+        return Err(Error::UnsupportedNtv2("Contains more than one subgrid"));
+    }
+
+    let (header, grid) = read_ntv2_subgrid(view, NTV2_HEADER_SIZE, is_le)?;
+    into_gravsoft_bin(header, grid)
+}
+
+/// Converts the `NTv2Grid` into binary gravsoft compatible with Rust Geodesy
+/// The gravsoft reader in geodesy_rs expects a text buffer made up of rows with f64 values delimited by spaces.
+/// Rows are ordered North to South and East to West.
+/// The first row contains the header values which should be in degrees
+/// The subsequent rows contain the grid values which should be in seconds or arc.
+/// See [geodesy_rs::grid::gravsoft_grid_reader](https://github.com/Rennzie/geodesy/blob/49384de0c70135fceac6f00ca367d171a1a8fe2e/src/grid/mod.rs#L208)
+fn into_gravsoft_bin(header: Vec<f64>, grid: Vec<Vec<f64>>) -> Result<Vec<u8>> {
+    let mut gravsoft = String::with_capacity(header.len() * 10 + grid.len() * grid[0].len() * 10);
+    for value in header {
+        gravsoft.push_str(&value.to_string());
+        gravsoft.push(' ');
+    }
+    gravsoft.push('\n');
+    for row in &grid {
+        for value in row {
+            gravsoft.push_str(&value.to_string());
+            gravsoft.push(' ');
+        }
+        gravsoft.push('\n');
+    }
+
+    Ok(gravsoft.into_bytes())
+}
+
+fn read_ntv2_subgrid(
```

Comment:
I feel so close, but testing with the pipeline
```bash
+proj=pipeline
  +step +inv +proj=tmerc +lat_0=49 +lon_0=-2 +k_0=0.9996012717 +x_0=400000 +y_0=-100000 +ellps=airy
  +step +proj=gridshift +grids=OSTN15_NTv2_OSGBtoETRS.gsb
  +step +proj=webmerc +lat_0=0 +lon_0=0 +x_0=0 +y_0=0 +ellps=WGS84

# Modifications:
# - tmerc: k changed to k_0
# - hgridshift changed to gridshift


# using controls points

 [
   544748.5367636156, 258372.49178149243, 9.61,
   544750.8907704452, 258365.94195330486, 9.61,
 ]
```

I'm getting a difference between PROJ and Geodesy as below

```bash
Converted Coords
13186.38247766841 6837193.773908622 9.61
13189.903112418948 6837182.964600426 9.61
Diff Expected Coords
182.0738776684102 -8.989791378378868 0
182.07421241894735 -8.997699573636055 0
```

The NON Gridshift version of the 27700 to 3857 re-projection has a negligible difference.

Reply:
After some more fiddling and now with a grid shape I trust to be correct. The difference between PROJ and Rust Geodesy for the pipeline above is

```bash
Diff Expected Coords
185.10826814375469 -79.77662696596235 0
185.10860513946682 -79.77751963492483 0
```